FrontEnd/src/LocationUtil.jsx

```diff
@@ -0,0 +1,27 @@
+import { database } from "./UserAuthentication/FirebaseConfig";
+import {collection} from "firebase/firestore"
+
+const DATABASE_FOLDER_NAME = "users";
+export const updateUserLocation = async (userUID, city, state) => {
+  try {
+    const userDocRef = collection(database, DATABASE_FOLDER_NAME, userUID);
+    await userDocRef.update({
+      UserLocation: `${city}, ${state}`,
+    });
+  } catch (error) {
+    throw new Error(`Unable to update user's location: ${error.message}`);
+  }
+};
+
+export const fetchAndUpdateUserLocation = async (userUID) => {
```

Comment:
I don't see this function updating user's location. Is this coming later?

Reply:
Yes, after the reverse geocoding